packages/connect-form/src/components/advanced-options-tabs/general/schema-input.tsx

```diff
@@ -0,0 +1,169 @@
+import { css } from '@emotion/css';
+import React from 'react';
+import {
+  Banner,
+  BannerVariant,
+  Description,
+  Label,
+  RadioBox,
+  RadioBoxGroup,
+  spacing,
+} from '@mongodb-js/compass-components';
+import ConnectionStringUrl from 'mongodb-connection-string-url';
+
+import { defaultHostname } from '../../../constants/default-connection';
+import { SetConnectionField } from '../../../hooks/use-connect-form';
+
+enum MONGODB_SCHEMA {
+  MONGODB = 'MONGODB',
+  MONGODB_SRV = 'MONGODB_SRV',
+}
+
+const descriptionStyles = css({
+  marginTop: spacing[1],
+});
+
+const regularSchemaDescription =
+  'Standard Connection String Format. The standard format of the MongoDB connection URI is used to connect to a MongoDB deployment: standalone, replica set, or a sharded cluster.';
+const srvSchemaDescription =
+  'DNS Seed List Connection Format. The +srv indicates to the client that the hostname that follows corresponds to a DNS SRV record.';
+
+function updateConnectionStringToStandard(
+  connectionStringUrl: ConnectionStringUrl
+): ConnectionStringUrl {
+  if (!connectionStringUrl.isSRV) {
+    // Already is standard schema, nothing to do.
+    return connectionStringUrl;
+  }
+
+  const newConnectionString = connectionStringUrl.toString();
+
+  const newConnectionStringUrl = new ConnectionStringUrl(
+    newConnectionString.replace('mongodb+srv://', 'mongodb://')
+  );
+
+  newConnectionStringUrl.hosts = [`${newConnectionStringUrl.hosts[0]}:27017`];
+
+  return newConnectionStringUrl;
+}
+
+function updateConnectionStringToSRV(
```

Comment:
We're doing these changes in this component currently. We could move this updating to the use-connect-form handler. Any preferences or other places? (This is done similarly in the host input as well.)

Reply:
I think this pattern works well with the typed actions we have for updating the form state that is invalid. I'm thinking for now we leave it but if, while adding more inputs to this form, we feel the side effects are hard to track lets go ahead and centralize them.

Reply:
Went ahead and centralized them in a change handler in `use-connect-form`. So each action in the field will go through this with its own action definition, and then it's own followable handling of that action. Should help get give us a better picture of various actions, which fields they impact, and in general keep things aligned I think. Thanks for bringing this up. Let me know if you're thinking we could do more here.